solr/core/src/java/org/apache/solr/core/SolrCores.java

```diff
@@ -95,6 +90,7 @@ protected void close() {
 
     // It might be possible for one of the cores to move from one list to another while we're
     // closing them. So loop through the lists until they're all empty. In particular, the core
+    // NOCOMMIT  I think i need help on determing what to do here.
```

Comment:
@dsmiley I am hoping you can weigh in on this.  Is there some code to be removed/simplified in a no transient core world?

Reply:
I suggest leaving this as-is in this PR.  I might change my mind if I dig into this more and want to change the abstractions in this PR... but I am not planning on it.